data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -546,6 +547,61 @@ private StateSyncCommitteesData getSyncCommitteesFromState(
             committeeIndices, spec.atEpoch(epoch).getConfig().getTargetCommitteeSize()));
   }
 
+  public SafeFuture<Optional<SyncCommitteeRewardData>> getSyncCommitteeRewardsFromBlockId(
+      final String blockId, final Set<String> validators) {
+    return getBlockAndMetaData(blockId)
+        .thenCompose(
+            result -> {
+              if (result.isEmpty() || result.get().getData().getBeaconBlock().isEmpty()) {
+                return SafeFuture.completedFuture(Optional.empty());
+              }
+              final BlockAndMetaData blockAndMetaData = result.get();
+              final BeaconBlock block = blockAndMetaData.getData().getBeaconBlock().get();
+              final SyncCommitteeRewardData data =
+                  new SyncCommitteeRewardData(
+                      blockAndMetaData.isExecutionOptimistic(), blockAndMetaData.isFinalized());
+
+              return combinedChainDataClient
+                  .getStateByBlockRoot(block.getRoot())
+                  .thenApply(
+                      maybeState ->
+                          getSyncCommitteeRewardData(validators, block, data, maybeState));
+            });
+  }
+
+  private Optional<SyncCommitteeRewardData> getSyncCommitteeRewardData(
+      Set<String> validators,
+      BeaconBlock block,
+      SyncCommitteeRewardData data,
+      Optional<tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState> maybeState) {
+    if (maybeState.isEmpty()) {
+      return Optional.empty();
+    }
+
+    final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state =
+        maybeState.get();
+
+    final UInt64 epoch = spec.computeEpochAtSlot(block.getSlot());
+    final UInt64 slot = spec.computeStartSlotAtEpoch(epoch);
+
+    final Optional<SyncCommittee> maybeCommittee =
+        spec.getSyncCommitteeUtil(slot).map(util -> util.getSyncCommittee(state, epoch));
```

Comment:
we can just use the slot of the block here, then delete the slot variable ...
or store the block.getSlot() as slot if you think that looks better, but ultimately no need to `spec.computeStartSlotAtEpoch`